### ISSUE TITLE
Array Support

### ISSUE THREAD

**Issue by rashtao** [arangodb/arangodb-spark-connector#17]
Tracked also here: https://github.com/arangodb/planning/issues/4217

The Spark Connector should support serializing / deserializing Arrays.
Relevant code should be here:

https://github.com/arangodb/arangodb-spark-connector/blob/025908e9d4ae324264df2b0103d2315d4d1ebc9a/src/main/scala/com/arangodb/spark/vpack/VPackUtils.scala#L75

https://github.com/arangodb/arangodb-spark-connector/blob/025908e9d4ae324264df2b0103d2315d4d1ebc9a/src/main/scala/com/arangodb/spark/vpack/VPackUtils.scala#L97

**Comment by fabiolab** [comment]
Hey,

Using savedf() to insert a DataFrame with ArrayType data into Arango makes this error occurs :

```
java.lang.ClassCastException: java.lang.String cannot be cast to scala.runtime.Nothing$
```
As mentioned bu @rashtao , this piece of code doesn't work since `getSeq(i)̀ needs to know the data type contained by the wrappedarray :

```
      case t: ArrayType => {
        builder.add(name, ValueType.ARRAY)
        row.getSeq(i).foreach { value => addValue(null, value, builder) }
        builder.close()
      }
```
For instance, for an array of String, this could work :
```
        row.getSeq[String](i).foreach { value => addValue(null, value, builder) }
```
However, I don't know how to fix the issue as transforming the dataframe into a RDD makes DataType information lost. 
How make it works for every data type ?

**Comment by fabiolab** [comment]
Ah, ok, datatype information is not lost. We just have to use the result of ```row.schema.fields```.

**Comment by rashtao** [comment]
fixed in v1.0.12
